src/Storages/System/StorageSystemDisks.cpp

```diff
@@ -44,14 +44,28 @@ Pipe StorageSystemDisks::read(
     MutableColumnPtr col_keep = ColumnUInt64::create();
     MutableColumnPtr col_type = ColumnString::create();
 
+    auto disk_type_to_string = [](DiskType disk_type)
+    {
+        switch (disk_type)
+        {
+            case DiskType::Local:
+                return "local";
+            case DiskType::RAM:
+                return "memory";
+            case DiskType::S3:
+                return "s3";
+        }
+        return "unknown"; // to make gcc10 happy
```

Comment:
Let's throw a `LOGICAL_ERROR` in such case. Because if we add new option to `DiskType` we need add it here to.
Also maybe it's better to move this function to same place where `DiskType` is defined to not forget change it if `DiskType` is changed.

Reply:
Do we want to convert the enum to struct with enum and `toString` function, like in `struct AccessRightsElementWithOptions`?

Reply:
You can try any variant and choose simplest one.

Reply:
Also you can write `__builtin_unreachable()`, see the examples in code.